File: files-ui/test_forms_and_streams/FormsAndStreams.py
#! python3
import Eto.Forms
import Eto.Drawing
import Rhino
import test

Rhino.RhinoApp.ClearCommandHistoryWindow()

test.test_print()
print("TEST")

counter = 0

def handler(s, e):
    global counter
    print(f"py3 test: {counter}")
    counter += 1

form = Eto.Forms.Form()
form.Owner = Rhino.UI.RhinoEtoApp.MainWindow
form.Size = Eto.Drawing.Size(400, 400)
form.Title = "Python 3"
form.MouseMove += handler 
form.Show()

File: files-ui/test_forms_and_streams/FormsAndStreams.cs
// #! csharp
using System;
using Eto.Forms;
using Eto.Drawing;
using Rhino;

Rhino.RhinoApp.ClearCommandHistoryWindow();

Test.Print();
Console.WriteLine("TEST");

int counter = 0;

var form = new Eto.Forms.Form();
form.Owner = Rhino.UI.RhinoEtoApp.MainWindow;
form.Size = new Eto.Drawing.Size(400, 400);
form.Title = "CSharp";
form.MouseMove += (s, e) => {
    Console.WriteLine($"csharp test: {counter}");
    counter++;
};
form.Show();


static class Test
{
    public static void Print() => Console.WriteLine("Test.Print");
}